pkg/apis/mattermost/v1alpha1/clusterinstallation_utils.go

```diff
@@ -324,17 +325,42 @@ func (mattermost *ClusterInstallation) GenerateDeployment(deploymentName, ingres
 				Key: "DB_CONNECTION_STRING",
 			},
 		}
+
+		if dbInfo.HasDatabaseCheckURL() {
+			initContainers = append(initContainers, corev1.Container{
+				Name:            "init-check-database",
+				Image:           "appropriate/curl:latest",
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Env: []corev1.EnvVar{
+					{
+						Name: "DB_CONNECTION_CHECK_URL",
+						ValueFrom: &corev1.EnvVarSource{
+							SecretKeyRef: &corev1.SecretKeySelector{
+								LocalObjectReference: corev1.LocalObjectReference{
+									Name: mattermost.Spec.Database.Secret,
+								},
+								Key: "DB_CONNECTION_CHECK_URL",
+							},
+						},
+					},
+				},
+				Command: []string{
+					"sh", "-c",
+					"until curl --max-time 5 $DB_CONNECTION_CHECK_URL; do echo waiting for mysql; sleep 5; done;",
+				},
+			})
+		}
 	} else {
 		masterDBEnvVar.Value = fmt.Sprintf(
```

Comment:
in the future, we should move this to a secret and inject that as env var

Reply:
Yep, I already have a ticket for this: https://mattermost.atlassian.net/browse/MM-26742